idaes/core/unit_model.py

```diff
@@ -590,8 +604,60 @@ def _get_stream_table_contents(self, time_point=0):
                     f"names (inet and outlet). Please contact the unit model "
                     f"developer to develop a unit specific stream table.")
 
-    def initialize(blk, state_args=None, outlvl=idaeslog.NOTSET,
-                   solver=None, optarg=None):
+    def initialize(blk, *args, **kwargs):
+        """
+        Initialization routine for Unit Model objects and associated
+        components.
+
+        This method is intended for initializing IDAES unit models and any
+        modeling components attached to them, such as costing blocks. First,
+        any attached components are deactivated, after which the
+        initialize_unit method is called. Following this, the attached
+        components are reactivated and theri initialize method called.
+
+        Currently, this method supports the following attached components:
+            * UnitModelCostingBlocks
+
+        Args:
+            costing_args - dict oarguments to be passed to costing block
+                           initialize method
+
+        For other arguments, see the initilize_unit method.
+        """
+        # Get any arguments for costing if provided
+        cost_args = kwargs.pop("costing_args", {})
+
+        # Get the costing block if present
+        # TODO: Clean up in IDAES v2.0
+        if hasattr(blk, "costing"):
+            costing = blk.costing
+        else:
+            costing = blk._costing_block_ref
```

Comment:
Shouldn't this be the other way around?  That is, the required name `_costing_ref` should be the definitive pointer to the costing block, and fall back on the old-style `costing` if it's not there?